packages/vscode-extension/src/handlers.ts

```diff
@@ -824,6 +824,35 @@ export async function runCommand(
   return result;
 }
 
+export async function downloadSampleApp(...args: unknown[]) {
+  const sampleId = args[1] as string;
+  const props: any = {
+    [TelemetryProperty.TriggerFrom]: getTriggerFromProperty(args),
+    [TelemetryProperty.SampleAppName]: sampleId,
+  };
+  ExtTelemetry.sendTelemetryEvent(TelemetryEvent.DownloadSampleStart, props);
+  const inputs: Inputs = getSystemInputs();
+  inputs["samples"] = sampleId;
+  inputs.projectId = inputs.projectId ?? uuid.v4();
+
+  const res = await downloadSample(inputs);
+  if (inputs.projectId) {
+    props[TelemetryProperty.NewProjectId] = inputs.projectId;
+  }
+  if (res.isOk()) {
+    props[TelemetryProperty.Success] = TelemetrySuccess.Yes;
+    ExtTelemetry.sendTelemetryEvent(TelemetryEvent.DownloadSample, props);
+    if (isValidOfficeAddInProject((res.value as vscode.Uri).fsPath)) {
+      await openOfficeDevFolder(res.value, true);
```

Comment:
why check officeaddin project? Does Office agent leverage this "downloadSampleApp" function?

Reply:
This is the same codes moved from `controls/webviewPanel.ts`, I will remove this part, thanks.